### ISSUE TITLE
Strongly define and implement predict() semantics of ObjectDetection models

### ISSUE THREAD

**Issue by LukeWood** [keras-team/keras-cv#1580]
My goal is as follows:

```
model = keras_cv.models.RetinaNet.from_preset(
    "spinenet_imagenet_open_images",
    bounding_box_format='xywh'
)
filepath = tf.keras.utils.get_file(origin="https://i.imgur.com/9i63gLN.jpg")
image = keras.utils.load_img(filepath)
boxes = classifier.predict([image])

visualization.plot_bounding_box_gallery([image],  [boxes] bounding_box_format='xywh', rows=1, cols=1)
```

![](https://i.imgur.com/9i63gLN.jpg)

![](https://i.imgur.com/Req9kQi.jpg)

There are a few building blocks missing here:

- Resizing at inference time
- Decoding of predictions in predict() to provide class labels (perhaps, need to study feasibility a bit more)

**Comment by sachinprasadhs** [comment]
This feature is handled now in KerasHub with the preprocessing function.

`RetinaNet` with the new implementation is now part of [KerasHub](https://github.com/keras-team/keras-hub) a consolidated `KerasNLP` and `KerasCV` package.
The model weights are available in `Kaggle`, for details and usage refer https://www.kaggle.com/models/keras/retinanet

We will not be making any changes to the models in `KerasCV` which are made available in `KerasHub` or the APIs/Utils made available in `Keras`.
If you still face any issue, please file a new issue in [keras-hub](https://github.com/keras-team/keras-hub)
